dask/dataframe/core.py

```diff
@@ -3321,10 +3321,11 @@ def _validate_condition(self, cond):
             or is_dataframe_like(cond)
             or is_series_like(cond)
             or is_index_like(cond)
+            or callable(cond)
```

Comment:
We could check whether the result of the callable can be aligned, but no strong opinion
Otherwise lgtm

Reply:
I added the check, but if it breaks other tests, I'd rather remove it and not complicate things.